### DISCUSSION TITLE
如何post到非home page

### DISCUSSION THREAD

**Discussion by liyuhangLLL** [ppoffice/hexo-theme-icarus#1275] [Q&A]
我创建了一些新的nav bar，如何post到特定的nav bar条目中？谢谢

**Comment by JIeJaitt** [comment]
应该是靠设置`formatter`，具体的你可以研究一下作者的site分支看看

**Comment by JIeJaitt** [comment]
你可以试试这个方法：

如果你希望一篇文章只在 "Blog" 导航栏下显示而不在首页显示，可以使用 Hexo 的自定义布局和条件语句来实现。以下是一种可能的方法：

1. 在 Hexo 的源文件夹中，创建一个自定义布局文件。例如，你可以在 `source/_layouts` 目录下创建一个名为 `blog-post-layout.ejs` 的文件。在这个布局文件中，你可以定义你希望在 "Blog" 导航栏页面中显示的文章内容。

```ejs
<!-- source/_layouts/blog-post-layout.ejs -->
<h2><%- page.title %></h2>
<!-- 可以在这里添加其他文章相关的内容 -->
```

2. 在 Hexo 的源文件夹中，编辑你想要只在 "Blog" 导航栏下显示的文章的 Markdown 文件。在文章的 Front Matter 中，指定使用自定义的布局。

```markdown
---
title: Your Article Title
layout: blog-post-layout
---
```

3. 在 Hexo 的源文件夹中，修改首页的布局文件，以排除在 "Blog" 导航栏页面下显示的文章。在首页的布局文件（通常是 `source/_layouts/index.ejs`）中，你可以使用条件语句来检查当前页面是否是 "Blog" 导航栏页面，如果不是，则显示文章列表。

```ejs
<!-- source/_layouts/index.ejs -->
<% if (page.layout !== 'blog-post-layout') { %>
  <% page.posts.each(function(post) { %>
    <h2><a href="<%- url_for(post.path) %>"><%- post.title %></a></h2>
    <!-- 可以在这里添加其他文章相关的内容 -->
  <% }) %>
<% } %>
```

4. 生成和部署网站：执行 Hexo 的生成和部署命令，将更改应用到最终的静态网站中。

```bash
hexo generate
hexo deploy
```

现在，你的文章将只在 "Blog" 导航栏页面中显示，而不在首页显示。

请注意，上述步骤提供了一种实现的思路，并且可能需要根据你所使用的 Hexo 主题和个人配置来进行适当的调整和修改。

**Reply by liyuhangLLL** [reply]
感谢你。我最后在hexo 官方的issues里找到了简单的方法。
https://github.com/hexojs/hexo/issues/1077
